1154 Reformatting Documentation (#239)
casper-ecosystem/cep-78-enhanced-nft: README.md

The mode is passed in as a `u8` value to `nft_holder_mode` runtime argument.

#### Minting

The minting mode governs the behavior of contract when minting new tokens. The minting modality provides two options:

1. `Installer`: This mode restricts the ability to mint new NFT tokens only to the installing account of the NFT contract.
2. `Public`: This mode allows any account to mint NFT tokens.
3. `ACL`: This mode allows whitelisted accounts or contracts to mint NFT tokens.

This modality is an optional installation parameter and will default to the `Installer` mode if not provided. However, this
mode cannot be changed once the contract has been installed. The mode is set by passing a `u8` value to the `minting_mode` runtime argument.

| MintingMode | u8  |
| ----------- | --- |
| Installer   | 0   |
| Public      | 1   |
| ACL         | 2   |

#### WhitelistMode

The `WhitelistMode` dictates if the ACL whitelist restricting access to the mint entry point can be updated. There are currently two options:

1. `Unlocked`: The ACL whitelist is unlocked and can be updated via the set variables endpoint.
2. `Locked`: The ACL whitelist is locked and cannot be updated further.

> If the `WhitelistMode` is set to `Locked` an ACL whitelist of entity keys must be provided on installation. This whitelist dictates which entities can mint NFTs in the restricted `ACL` minting mode. These entities include `Accounts` and/or `Contracts`.

Reviewer: The information about ACL didn't transfer over to the modalities.md file.

Reviewer 2: Good catch! Not sure what was going on there, but it's updated now.

The modality provides three options:

1. `NoEvents`: This modality will signal the contract to not record events at all. This is the default mode.
2. `CEP47`: This modality will signal the contract to record events using the CEP47 event schema. Further information can be found [below](#cep47-mode).
3. `CES`: This modality will signal the contract to record events using the [Casper Event Standard](#casper-event-standard).

| EventsMode | u8  |
| ---------- | --- |
| NoEvents   | 0   |
| CEP47      | 1   |
| CES        | 2   |

#### Transfer Filter Hook

The transfer filter modality, if enabled, specifies a contract package hash pointing to a contract that will be called when the `transfer` method is invoked on the contract. CEP-78 will call the `can_transfer`
method on the specified callback contract, which is expected to return a value of `TransferFilterContractResult`, represented as a u8.

- `TransferFilterContractResult::DenyTransfer` will block the transfer regardless of the outcome of other checks
- `TransferFilterContractResult::ProceedTransfer` will allow the transfer to proceed if other checks also pass

> The transfer filter can be enabled by passing a `ARG_TRANSFER_FILTER_CONTRACT` argument to the install method, with a value of type `Option<Key>`

Reviewer: The transfer filter hook also needs to move to the modalities.md file. Perhaps there was a merge issue?
Also, I notice that this modality does not follow the formatting of the previous ones that have a table and a list of options.

Reviewer 2: This is bizarre. I don't remember writing this or when it showed up in here - I'm going to have to take a look at the previous commits.

I'm fairly confident this wasn't in my local, updated cep-78 repo.

> ### Installing the Contract

Reviewer: Since we have another section for installing the contract with CLI, could we rename this to:
```suggestion
### Building the Contract
```

> The `main.rs` file within the contract provides the installer for the NFT contract. Users can compile the contract to Wasm using the `make build-contract` with the provided Makefile.

Reviewer: I know this is the original text, but I suggest a minor update here:
```suggestion
The `main.rs` file within the contract provides the installer for the NFT contract. Users can compile the contract to Wasm using the `make build-contract` command from the Makefile provided.
```

> The pre-built Wasm for the contract and all other utility session code can be found as part of the most current release. Users wishing to build the Wasm themselves can pull the code and the `make build-contract` provided in the included Makefile. Please note, however, that as part of building the contract, you will need to install `wasm-strip`.

Reviewer: ```suggestion
The pre-built Wasm for the contract and all other utility session code can be found as part of the most current release. Users wishing to build the Wasm themselves can pull the code and use the `make build-contract` command provided in the Makefile. Please note, however, that you must install `wasm-strip` to build the contract.
```

> The `call` method will install the contract with the necessary entrypoints and call the `init()` entrypoint to allow the contract to self initialize and setup the necessary state to allow for operation,

Reviewer: Missing colon and rewording:
```suggestion
The `call` method will install the contract with the necessary entrypoints and call the `init()` entrypoint, which allows the contract to self-initialize and set up the necessary state variables for operation.
```

> 

Reviewer: I recommend adding another section here:
```suggestion
### Required Runtime Arguments
```

> The following is an example of deploying the installation of the NFT contract via the Rust Casper command client.

Reviewer: It might be useful to point to the rest of the examples.
```suggestion
The following is an example of installing the NFT contract via a deploy using the Rust CLI Casper client. You can find more examples [here](/docs/using-casper-client.md).
```
